tests/test_hash.py

```diff
@@ -45,13 +45,89 @@
 
 class TestHash(unittest.TestCase):
 
-  def _run_with_all_hash_libraries(self, test_func):
-    for lib in securesystemslib.hash.SUPPORTED_LIBRARIES:
+  @staticmethod
+  def _get_algorithms(library):
+    algorithms = ['md5', 'sha1', 'sha224', 'sha256', 'sha384', 'sha512']
+    if library != 'pyca_crypto':
```

Comment:
I'm not sure if this should be so dynamic between every function call. Also, it assumes that the library has been/not been loaded/unloaded.